### DISCUSSION TITLE
Fix sharing of hidden albums

### DISCUSSION THREAD

**Discussion by ajsteele** [ente-io/ente#1571] [Enhancements]
If I share a hidden album with someone, it appears in their shared albums, unhidden. In fact, even worse, it’s not possible to hide shared albums so the recipient can’t do this manually if they notice the issue!

This is obviously very much not desired behaviour for potentially sensitive images so it would be great if it did the expected thing and it was possible to hide shared albums, and hidden albums were automatically shared hidden.

**Comment by vishnukvmd** [comment]
Hey, sorry about this!

Automatically hiding such albums on the receiver's device could cause confusion, and you'd have to communicate out of band to let them know where to find these items.

But we should definitely add an option for the receiver to hide shared albums at their end, so that they've an option to lock away sensitive items. For now, could you please use "Archive" option? This will not lock these items, but will hide them from the home gallery.

In the meanwhile, we'll figure out how we can prioritize the former. Thanks for the feedback!

**Reply by ajsteele** [reply]
Thanks for getting back to me. :)

I’m not sure which is the worse/more confusing behaviour—having to communicate out-of-band that they’ve got a new folder that’s already hidden, or having to communicate out-of-band to let them know to immediately hide the album you shared! I’d say the former is a safer option but am open to being convinced!

And thanks for suggesting archiving as a workaround!

**Reply by vishnukvmd** [reply]
There is this corner case of me not wanting a receiver to know that I've hidden a particular album.

Now this could be an awkward bit of information to communicate, depending on the intent with which the album was hidden (noisy vs. sensitive vs. a use case we haven't yet thought of).

There is no perfect answer, but just clarifying why we're currently leaning towards scoping "hidden" to a user, rather than making it a public property of an album :)

**Reply by ajsteele** [reply]
Could the best solution be a checkbox/slider when you share saying ‘also hide this album for recipient’? I guess I’d default to ‘yes’, but depending on how you weight the corner-case you might disagree! I think there might well be cases where people would want to be able to set this independently though, so scoping to user does make sense.

**Reply by mnvr** [reply]
IMO the real issue here is this:

> it’s not possible to hide shared albums so the recipient can’t do this manually if they notice the issue

There isn't any other issue (again, IMO). If I share an album with someone then they should not be privy to the fact if I have archived / hidden that album. But yes, they should be able to hide it on their own. My point being, I don't think we need new toggles etc, we just need to support hiding of shared albums.

**Reply by ajsteele** [reply]
I think the only issue is that they might _want_ it to be hidden and you might not care that they know, eg if it’s some joint private photos. It would be nice if this could be done seamlessly by allowing the sender to pre-hide or -archive the album.

btw we tried it and you don’t seem to be able to archive shared albums either!

**Reply by mnvr** [reply]
We understand about the use case you're describing, but we feel it is a use case tackled separately. Something like "partner sharing" or a slightly more generic variant of that, where both parties trust each other and to share and organize full library collectively. We do want to solve that use case too, but not by mixing it with the regular sharing flows, because 1. it'll complicate all the regular sharing flows, and 2. it'll still be inconvenient for folks who want partner sharing because each time they share they'll have to go to some advanced options or whatnot and toggle a couple of switches.

I don't know the final form partner sharing will take, and when we'll get around to it, but I just wanted to tell you folks in the team have been thinking about it for long and do want to tackle these cases (since many of us also want something similar for our own use!)

>  you don’t seem to be able to archive shared albums either!

That too is an infelicity we'll fix, same as being able to hide shared albums.

**Reply by pchampio** [reply]
Hello, I had the same expectation that default behavior when sharing a hidden album with someone was to have it in the hidden folder for the receiver.
The opposite behavior was not intuitive and felt like a bug.

**Reply by eagle-zero** [reply]
> There isn't any other issue (again, IMO). 

I highly disagree. When a hidden album is created, it has the function of preventing someone else from accidentally opening this album by an extra layer of authentication. If a hidden album with obviously sensitive photos is shared, then this is done with the intention of making these photos accessible to a specific person. So the user has decided to extend the access to the sensitive photos to that person. However, just because the photos are shared does not mean that these photos are no longer sensitive and should no longer protected from unwanted access by others.

**Reply by wideawake42** [reply]
Wanting to be able to hide albums of sensitive material that are shared, and hoping this gets implemented soon.

I don't need auto hide and think that if you don't trust a person to keep an album hidden, well, t's not that hard for them to just unlock their hidden files and show other people, and so you shouldn't be sharing with people you don't trust in the first place. You should communicate when you share things that should be hidden and agree on it ahead of time and then they can hide it when you send it.

But they need to be able to hide it as they decide. It's not just about lewds, it's also about things like a person's individual comfort level. E.g. if I have photos of someone's injury that are a bit gruesome and my husband needs access to those photos but feels squeamish about having them in his timeline all the time, it would be nice for him to be able to hide them when not needed. Archiving them removes them from the timeline but you can still see them in the albums list, and since they have a thumbnail, it gives the album away unless you stick in something unrelated.

Currently Ente is great for sharing photos securely except for this (well, and my other gripe is that I have to have an overflow album for photos that are in shared albums I don't own or else they just clutter up my "uncategorized" -- but that's less pressing).

**Comment by anandbaburajan** [comment]
Fixed in https://github.com/ente-io/ente/pull/8842 and https://github.com/ente-io/ente/pull/8846
